src/op_handlers/log.rs

```diff
@@ -3,78 +3,75 @@ use u256::U256;
 
 pub fn storage_write(
     vm: &mut VMState,
     opcode: &Opcode,
-    state_storage: &mut StateStorage,
+    world: &mut World,
 ) -> Result<(), EraVmError> {
     let key_for_contract_storage = vm.get_register(opcode.src0_index).value;
     let address = vm.current_context()?.contract_address;
     let key = StorageKey::new(address, key_for_contract_storage);
     let value = vm.get_register(opcode.src1_index).value;
-    state_storage.storage_write(key, value)?;
+    world.storage_write(key, value);
     Ok(())
 }
 
-pub fn storage_read(
-    vm: &mut VMState,
-    opcode: &Opcode,
-    state_storage: &StateStorage,
-) -> Result<(), EraVmError> {
+pub fn storage_read(vm: &mut VMState, opcode: &Opcode, world: &World) -> Result<(), EraVmError> {
     let key_for_contract_storage = vm.get_register(opcode.src0_index).value;
     let address = vm.current_context()?.contract_address;
     let key = StorageKey::new(address, key_for_contract_storage);
-    let value = state_storage.storage_read(key)?.unwrap_or(U256::zero());
+    let value = world.storage_read(key)?.unwrap_or(U256::zero());
     vm.set_register(opcode.dst0_index, TaggedValue::new_raw_integer(value));
     Ok(())
 }
 
 pub fn transient_storage_write(
     vm: &mut VMState,
     opcode: &Opcode,
-    transient_storage: &mut StateStorage,
+    world: &mut World,
 ) -> Result<(), EraVmError> {
     let key_for_contract_storage = vm.get_register(opcode.src0_index).value;
     let address = vm.current_context()?.contract_address;
     let key = StorageKey::new(address, key_for_contract_storage);
     let value = vm.get_register(opcode.src1_index).value;
-    transient_storage.storage_write(key, value)?;
+    world.transient_storage_write(key, value);
     Ok(())
 }
 
 pub fn transient_storage_read(
     vm: &mut VMState,
     opcode: &Opcode,
-    transient_storage: &StateStorage,
+    world: &World,
 ) -> Result<(), EraVmError> {
     let key_for_contract_storage = vm.get_register(opcode.src0_index).value;
     let address = vm.current_context()?.contract_address;
     let key = StorageKey::new(address, key_for_contract_storage);
-    let value = transient_storage.storage_read(key)?.unwrap_or(U256::zero());
+    let value = world.transient_storage_read(key)?.unwrap_or(U256::zero());
     vm.set_register(opcode.dst0_index, TaggedValue::new_raw_integer(value));
     Ok(())
 }
 
 pub fn add_l2_to_l1_message(
     vm_state: &mut VMState,
     opcode: &Opcode,
-    state_storage: &mut StateStorage,
+    world: &mut World,
 ) -> Result<(), EraVmError> {
     let key = vm_state.get_register(opcode.src0_index).value;
     let value = vm_state.get_register(opcode.src1_index).value;
     let is_service = opcode.imm0 == 1;
-    state_storage.record_l2_to_l1_log(L2ToL1Log {
+    world.record_l2_to_l1_log(L2ToL1Log {
         key,
         value,
         is_service,
         address: vm_state.current_context()?.contract_address,
         shard_id: 0,
         tx_number: vm_state.tx_number as u16,
```

Comment:
Not from this PR, but, why tx_number is u16 but the VM struct is u64? This can panic, how are we checking this cast is safe?

Reply:
Mmm... I don't know about that, maybe it is worth opening an issue.

Reply:
[Sure](https://github.com/lambdaclass/era_vm/issues/198)